File: genom.go
package main

import (
	"fmt"
)

func genome() {
	var str1, str2 string
	genom1 := NewStringSet()
	genom2 := NewStringSet()

	fmt.Println("Введите геномы")
	fmt.Scan(&str1)
	fmt.Scan(&str2)

	for i := 0; i < len(str1)-1; i++ {
		addElem := string(str1[i]) + string(str1[i+1])
		genom1.Add(addElem)
	}

	for i := 0; i < len(str2)-1; i++ {
		addElem := string(str2[i]) + string(str2[i+1])
		genom2.Add(addElem)
	}

	intersectionGenom := genom1.IntersectionWith(genom2)
	fmt.Printf("Similarity = %d\n", intersectionGenom.Size())
}

File: genom.cpp
#include <iostream>
#include "setInclude.h"

using namespace std;


void genome(){
    string str1;
    string str2;
    StringSet genom1;
    StringSet genom2;
    cout << "Введите геномы" << endl;
    cin >> str1;
    cin >> str2;
    for (size_t i = 0; i < str1.size() - 1; i++){
        string addElem = "";
        addElem += str1[i];
        addElem += str1[i+1];
        genom1.add(addElem);
    }
    for (size_t i = 0; i < str2.size() - 1; i++){
        string addElem = "";
        addElem += str2[i];
        addElem += str2[i+1];
        genom2.add(addElem);
    }
    StringSet intersectionGenom = genom1.intersectionWith(genom2);
    cout << "Similarity = " << intersectionGenom.size() << endl;
}